hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -4212,6 +4218,48 @@ func (r *HostedClusterReconciler) serviceAccountSigningKeyBytes(ctx context.Cont
 	return privateKeyPEMBytes, publicKeyPEMBytes, nil
 }
 
+func (r *HostedClusterReconciler) reconcileKubevirtPlatformDefaultSettings(ctx context.Context, hc *hyperv1.HostedCluster) error {
+	// auto generate the basedomain by retrieving the default ingress *.apps dns.
+	baseDomainPassthrough := hc.Annotations[hyperv1.KubeVirtBaseDomainPassthroughAnnotation]
+	if baseDomainPassthrough == "true" && hc.Spec.DNS.BaseDomain == "" {
```

Comment:
what happens if `baseDomainPassthrough == "true" && hc.Spec.DNS.BaseDomain =!= ""`, Is that valid?

I'm +1 to make this a field and the include any invalid input combination for .baseDomainPassthrough and .BaseDomain in to `validateConfigAndClusterCapabilities`

Reply:
That scenario is not allowed during Create. After creation, the hostedcluster controller updates the BaseDomain field to the correct value based on domain name of the cluster running the VMs. 

I added a creation check `validateKubevirtBaseDomainPassthroughCreate` that ensures someone can't try to both enable the passthrough feature and set their own basedomain. That check is a part of the webhook, so it might be best effort, but i think that's probably the most feasible option for catching this configuration error.